api/scm.go

```diff
@@ -164,14 +169,50 @@ func SyncRepo(c *gin.Context) {
 		// set repo to inactive - do not delete
 		r.SetActive(false)
 		// update repo in database
-		e := database.FromContext(c).UpdateRepo(r)
-		if e != nil {
+		err := database.FromContext(c).UpdateRepo(r)
+		if err != nil {
 			retErr := fmt.Errorf("unable to update repo for org %s: %w", org, err)
 
 			util.HandleError(c, http.StatusInternalServerError, retErr)
 
 			return
 		}
+		err = updateUserFavorites(c, r)
+		if err != nil {
+			util.HandleError(c, http.StatusInternalServerError, err)
+			return
+		}
 	}
 	c.JSON(http.StatusOK, fmt.Sprintf("repo %s synced", r.GetFullName()))
 }
+
+// updateUserFavorites is a helper function that takes a repo
+// that has been deleted from the SCM and ensures that it does
+// not appear in the user's favorites after its deletion.
+func updateUserFavorites(c *gin.Context, r *library.Repo) error {
+	// get user from the database
+	u, err := database.FromContext(c).GetUser(r.GetUserID())
+	if err != nil {
+		retErr := fmt.Errorf("%s: failed to retrieve user in database", baseErr)
+		util.HandleError(c, http.StatusBadRequest, retErr)
```

Comment:
AFAICT, if this function returns an `error`, then we call `util.HandleError()` with that error:

https://github.com/go-vela/server/blob/20f7ffd20dcbb07f207227ae46a401b3a84e6176/api/scm.go#L180-L184

AFAIK, that function will "abort" the current request preventing any subsequent calls from proceeding:

https://github.com/go-vela/server/blob/master/util/util.go#L18

https://pkg.go.dev/github.com/gin-gonic/gin#Context.AbortWithStatusJSON

I mention all of this, because I think we can remove this line.